src/kakarot/instructions/system_operations.cairo

```diff
@@ -505,11 +518,16 @@ namespace CreateHelper {
 
         // Load bytecode code from memory
         let (bytecode: felt*) = alloc();
-        let memory = Memory.load_n(
+        let (memory, gas_cost) = Memory.expand_and_load_n(
             self=ctx.memory, element_len=size, element=bytecode, offset=offset
         );
         let ctx = ExecutionContext.update_memory(ctx, memory);
 
+        // code_deposit_cost = 200 * deployed_code_size
+        // TODO: update dynamic_gas to be dynamic_gas := memory_expansion_cost + deployment_code_execution_cost + code_deposit_cost
```

Comment:
Can’t this be done in `finalize` ?

Reply:
finalize supposes we have gotten to the end of the execution, right?
i'm guessing these costs are incurred at the beginning of the execution

Reply:
To the end of the execution of this opcode